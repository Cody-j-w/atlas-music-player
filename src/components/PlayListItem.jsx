export default function PlayListItem(props) {
    return (
        <div className="flex justify-between my-3 hover:bg-bright-red group">
            <div>
                <p className="font-semibold text-lg dark:text-bright-red text-charcoal ">{props.songTitle}</p>
                <p className="font-semibold text-md dark:text-silver text-charcoal ">{props.artist}</p>
            </div>
            <div>
                <p className="font-semibold text-md dark:text-silver text-charcoal ">{props.songLength}</p>
            </div>
        </div>
    )
}