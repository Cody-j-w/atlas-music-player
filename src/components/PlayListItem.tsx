export default function PlayListItem(props: { songTitle: string, artist: string, songLength: string }) {
    return (
        <div className="flex justify-between my-3 hover:bg-bright-red group">
            <div>
                <p className="font-semibold text-lg dark:text-bright-red text-charcoal dark:group-hover:text-charcoal ">{props.songTitle}</p>
                <p className="font-semibold text-md dark:text-silver text-charcoal dark:group-hover:text-charcoal ">{props.artist}</p>
            </div>
            <div>
                <p className="font-semibold text-md dark:text-silver text-charcoal dark:group-hover:text-charcoal ">{props.songLength}</p>
            </div>
        </div>
    )
}