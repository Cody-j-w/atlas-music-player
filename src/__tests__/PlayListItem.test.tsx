import PlayListItem from "../components/PlayListItem";
import { test, expect } from "vitest";
import { render } from "@testing-library/react";

test('SongTitle snapshot', () => {
    const testSet = (currentTrack: string) => {
        return;
    }

    const { container } = render(<PlayListItem songTitle="test One" artist="The Tests" songId="1234" currentTrack="1234" setCurrentTrack={testSet} duration={194} />);
    expect(container).toMatchSnapshot();
});