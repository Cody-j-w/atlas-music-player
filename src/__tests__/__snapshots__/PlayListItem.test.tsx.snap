// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CoverArt snapshot 1`] = `
<div>
  <div
    class="flex justify-between my-3 hover:bg-bright-red group border-2 p-2 border-silver text-chrome-gold"
  >
    <div>
      <p
        class="font-semibold text-lg bg-charcoal text-chrome-gold"
      >
        test One
      </p>
      <p
        class="font-semibold text-lg bg-charcoal text-chrome-gold"
      >
        The Tests
      </p>
    </div>
    <div>
      <p
        class="font-semibold text-lg bg-charcoal text-chrome-gold"
      >
        3:14
      </p>
    </div>
  </div>
</div>
`;

exports[`SOngTitle snapshot 1`] = `
<div>
  <div
    class="flex justify-between my-3 hover:bg-bright-red group border-2 p-2 border-silver text-chrome-gold"
  >
    <div>
      <p
        class="font-semibold text-lg bg-charcoal text-chrome-gold"
      >
        test One
      </p>
      <p
        class="font-semibold text-lg bg-charcoal text-chrome-gold"
      >
        The Tests
      </p>
    </div>
    <div>
      <p
        class="font-semibold text-lg bg-charcoal text-chrome-gold"
      >
        3:14
      </p>
    </div>
  </div>
</div>
`;

exports[`SongTitle snapshot 1`] = `
<div>
  <div
    class="flex justify-between my-3 hover:bg-bright-red group border-2 p-2 border-silver text-chrome-gold"
  >
    <div>
      <p
        class="font-semibold text-lg bg-charcoal text-chrome-gold"
      >
        test One
      </p>
      <p
        class="font-semibold text-lg bg-charcoal text-chrome-gold"
      >
        The Tests
      </p>
    </div>
    <div>
      <p
        class="font-semibold text-lg bg-charcoal text-chrome-gold"
      >
        3:14
      </p>
    </div>
  </div>
</div>
`;
